src/EditorFeatures/CSharpTest/ConvertAutoPropertyToFullProperty/ConvertAutoPropertyToFullPropertyTests.cs

```diff
@@ -858,6 +858,50 @@ class TestClass
             await TestMissingAsync(text);
         }
 
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task CursorInType()
+        {
+            var text = @"
+class TestClass
+{
+    public int[||] Goo { get; set; }
```

Comment:
This a breaking change, previously it registered [anywhere](https://github.com/dotnet/roslyn/pull/36879/files#diff-9267bfd79b8e7cfbf98b95ee077cfa34L50) in signature. Again, I'm not sure if its desired (I'd say so) so let's have a discussion about it.

Reply:
Yes.  this seems like an unfortunate regression. This fits into the "is it in the header" logic that i think we should have.

For members/types the header usually extends from the start of the modifiers to the 'body' (where body is weakly, but somehwat intuitively defined).  We can expand that out per construct if you want.

Reply:
I feel like that could conceptually be part of the extractor. It extracts parent node from current header node. One issue is that it's in a way parallel to the IsOnHeader approach that already exists. On the other hand we went it very opposite approach in the helpers (`Give me node` instead of `Am I on relevant node` so I don't think we have all that much choice ).

Reply:
Also, it brings a few more interesting questions. If we enable this user might expect method's refactorings to show up when caret is on return type as well. And that one might a more complicated (i.e. with its own set of refactorings). Do we want to enable it there as well?

Reply:
Yes.  I would have it be available there as this is "part of the header/signature" for a user.  In general, this isn't problematic as types are not *normally* arbitrarily large.  so it still feels like "you are on hte property" when you're on such a type.  Contrast that with nested statements/expressions/bodies/etc. where it very quickly doesn't feel like you're "on" the parent construct anymore.